### ISSUE TITLE
unless with a test as context

### ISSUE THREAD

**Issue by GuillaumeBiton** [nolimits4web/Template7#4]
Hello,

Can I use a test function as context for #unless helper. I have read in source that a test on context is made at the top of the function but I can make it work.

    {{#unless function(){ return var === "test";} }}
        ...
    {{/unless}}

maybe it could be easier to do something like that :

    {{#unless (var === test) }}
        ...
    {{/unless}}

Regards

**Comment by nolimits4web** [comment]
Hi there,
It would be really hard to parse such conditions, but you can just create your custom helper, like i did for myself with `if`
```
Template7.registerHelper('if_compare', function (a, operator, b, options) {
    var match = false;
    if (
        (operator === '==' && a == b) ||
        (operator === '===' && a === b) ||
        (operator === '!=' && a != b) ||
        (operator === '>' && a > b) ||
        (operator === '<' && a < b) ||
        (operator === '>=' && a >= b) ||
        (operator === '<=' && a <= b)
        ) {
        match = true;
    }
    if (match) return options.fn(this);
    else return options.inverse(this);
});
```
And then use it like:
```
{{#if_compare a ">" b}} ... {{/if_compare}}
{{#if_compare a "===" b}} ... {{else}} ... {{/if_compare}}
```
So you can do the same for `unless` helper

**Comment by GuillaumeBiton** [comment]
OK thanks vladimir, I will do it

**Comment by GuillaumeBiton** [comment]
Hello again,

Sorry but It won't work :

    T7.registerHelper('unmatch', function (a, b, options) {
		if (a !== b) {
			return options.fn(this, options.data);
		} else {
			return options.inverse(this, options.data);
		}
	});

test is ok but nothing is hide by Template7

**Comment by GuillaumeBiton** [comment]
Ok my bad, I wasn't closing correctly my helper in template page

**Comment by MarioVanDenEijnde** [comment]
Hello Framework team,

I want to pass a json object (or json string) from page to page from inside a template
I tried the following but do not succeed:

Below we call the template and present the json object.
We also tried to stringify the json but also with no result.

Can you advise please?
```
}).then(function(result){
	console.log(JSON.stringify(result[0]));
	Template7.registerHelper('stringify', function (context){ return JSON.stringify(context[0]); });
	var foodsearchHTML = $('script#foodsearchresultTemplate').html();
	var compiledFoodSearchTemplate = Template7.compile(foodsearchHTML);
	foodHTML3 = compiledFoodSearchTemplate(result);
	document.getElementById('content-wrap-foodsearch').innerHTML = foodHTML3;
});
```
The template is as follows:
```
<script id="foodsearchresultTemplate" type="text/template7">
<div class="list-block media-list">
<ul>
{{#each this}}
    <li>
	  <a onclick="open7consumptionNew('{{#stringify this}}');" data-context-name="object.{{this}}" class="item-content item-link">
        <!--<div class="item-media"><i class="icon icon-f7"></i></div>-->
        <div class="item-inner">
        <div class="row">
          <div class="item-title col-70" style="padding-left:10px">{{this.title}}</div>
          <div class="item-after col-30">{{this.unitNameFriendly}}</div>
        </div>
        </div>
      </a>
    </li>
{{/each}}
</ul>
</div>
</script>
```
